src/analyser/encoding.zig

```diff
@@ -146,7 +146,7 @@ pub fn decode(extra: *[]const u8, comptime T: type) T {
         },
         .Array => |info| blk: {
             var array: T = undefined;
-            var i: usize = 0;
+            const i: usize = 0;
             while (i < info.len) {
                 array[i] = decode(extra, info.child);
             }
```

Comment:
interesting :eyes:

Reply:
Would you like me to fix it in this PR or leave it to you to handle in the "intern pool rewrite" PR?

Reply:
My IP work will take some time so I would suggest to fix it in this PR. Should be quite obvious what to do.